backend/app/poller.py

```
@@ -142,36 +141,34 @@ async def send_push_notifications(notifications, device_tokens, loop):
-            ))
+            subscription_info = json.loads(token)
+            messages.append((subscription_info, json.dumps(data_payload)))
 
     if not messages: return
 
     for i in range(0, len(messages), 10):
         chunk = messages[i:i + 10]
         try:
             logging.info(f"[FCM] Sending a chunk of {len(chunk)} messages...")
-            response = await loop.run_in_executor(None, lambda: messaging.send_each(chunk))
+            response = await loop.run_in_executor(None, lambda: [webpush(si, data, VAPID_PRIVATE_KEY_FILE, VAPID_CLAIMS) for si, data in chunk])
```

Ruff (0.15.20)

[warning] Function definition does not bind loop variable `chunk`

(B023)